presto-hive/src/main/java/com/facebook/presto/hive/ColumnarBinaryHiveRecordCursor.java

```diff
@@ -452,11 +458,30 @@ else if (hiveTypes[column].equals(HIVE_LONG)) {
                 longs[column] = readVInt(bytes, start, length);
             }
         }
+        else if (hiveTypes[column].equals(HIVE_FLOAT)) {
+            checkState(length >= 1, "Float should be at least 1 byte");
+            if (length == 1) {
+                longs[column] = bytes[start];
+            }
+            else {
+                longs[column] = readRawBits(bytes, start, length);
```

Comment:
Fixed in c51c067279091ff3be3b60fd50c2cfa5c5701d05.

BTW I remember I had once some issues with Float column having less than 4 bytes in size. I cannot remember what were steps to reproduce. I double checked and Apache manual says float in Hive is 4 byte in size though.